ckanext/datastore/writer.py

```diff
@@ -11,6 +12,14 @@
 
 BOM = "\N{bom}"
 
+# (canada fork only): conform to XML stadards for element names
+# TODO: upstream contrib
+xml_element_name_rules = [
+    (re.compile(r'^([0-9]+xml|xml[0-9]+|xml|^[0-9]+)', re.I), ''),
+    (re.compile(r'\ +'), '_'),
```

Comment:
space needs escaping? Maybe only with some of the wackier regex options. Also `\s` might be better than just the space character here.